### PR TITLE
lib: Move ServerTime component to PF4 TimePicker

### PR DESCRIPTION
**Pull request by martinpitt**
This brings it in line with the shutdown and Services timer dialog, and
avoids an unsightly dialog layout on mobile.

Change the state keeping to just have a single tiem string as variable,
as that's overall easier to manage. Move the minute normalization into
get_current_time(), as that's the only place where it's really necessary
-- writing out the values does not care about leading zeros.

Add a pixel test for the manual time setting. This does not capture the
whole expanded time picker list, as that grows beyond the dialog, but
still captures enough to make sure it looks as expected. There is no way
to collapse it again, but we want to do the screenshot after setting a
manual date and time, to include these input value strings into the
pixel test.

----

In main, the mobile dialog looks like this, with an unhappy break:

manual-time-mobile

See the pixel tests for how the dialog looks like now:

desktop

mobile

### CODE REVIEW

**Comment by martinpitt** [comment]
[This](https://logs.cockpit-project.org/logs/pull-16395-20210924-104951-aca2dba8-fedora-34-firefox/log.html#287-1) is an unrelated failure, but the [mobile pixel test](https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/pull-16395-20210924-104951-aca2dba8-fedora-34-mobile/pixeldiff.html#TestSystemInfo-testTime-systime-manual-time-mobile) has a subtle rendering error for the clock icon. I did these screenshots with the exact same chromium version (in the cockpit/tasks container), so I figure I ignore that icon for the time being.

**Comment by martinpitt** [comment]
I added another "dialog settles down" waiting which hopefully works, and also added a fix for [this flake](https://logs.cockpit-project.org/logs/pull-16395-20210924-104951-aca2dba8-fedora-34-firefox/log.html#287-1)

**Comment by martinpitt** [comment]
Eh, what? The [failed pixel text ](https://logs.cockpit-project.org/logs/pull-16395-20210924-125820-9fca2ba3-fedora-34-mobile/pixeldiff.html#TestSystemInfo-testTime-systime-manual-time-mobilehttps://logs.cockpit-project.org/logs/pull-16395-20210924-125820-9fca2ba3-fedora-34-mobile/TestSystemInfo-testTime-systime-manual-time-mobile-pixels.png) shows an alleged mobile screenshot which is *most definitively not* mobile -- that's desktop. This works fine locally..

**Comment by martinpitt** [comment]
I can reproduce [this pixel test failure](https://logs.cockpit-project.org/logs/pull-16395-20210927-052033-9fca2ba3-fedora-34-mobile/pixeldiff.html#TestSystemInfo-testTime-systime-manual-time-mobile) with an extra `time.sleep(0.5)` in the test -- it seems the animation is not completely done yet in mine. As there is no DOM change at all, I can really only think of a sleep :cry:

**Comment by martinpitt** [comment]
I give up on the [icon jitter on mobile](https://logs.cockpit-project.org/logs/pull-16395-20210927-073617-0bae9c08-fedora-34-mobile/log.html#291) -- I changed the test to ignore the `<input>` on mobile (which has that icon). That'll still validate the overall layout, as the frame around it is a parent element already.

**Comment by KKoukiou** [comment]
@martinpitt sorry it's conflicty now

**Comment by martinpitt** [comment]
Conflicts after merging #16389, rebasing. I'm not sure if the screenshots are affected, I'll let the bots decide

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**
